hist/histv7/inc/ROOT/RAxis.hxx

```diff
@@ -759,6 +790,46 @@ public:
          vec.at(kv.second) = kv.first;
       return vec;
    }
+
+   /// Result of an RAxisLabels label comparison
+   enum LabelComparisonFlags {
```

Comment:
`enum class`? It's nice to have `kSame` prefixed with what's same.

Reply:
I'm not sure...

- `enum class` is not widely used in ROOT (even v7), so this may stand out and feel verbose.
    * And until ROOT switches to C++20, which _finally_ introduces `using enum`, there will be no good way to reduce the verbosity.
- Bit manipulations will cause lots of casting into int and back, obscuring the code.
- ...which is arguably because overall, using enums for bitflags is really a hack, and `enum class` makes the hack less convenient to use.

So, why I'm doing it if I consider it to be a hack? Because that just happens to be the standard hack for bitflags in the C family, as all other options are either much more syntax-heavy (like the "newtype + one const per flag approach") or problematic in other ways (like bitfields).

Reply:
Also, people who want to clarify what's same can just write `RAxisLabels::LabelsComparisonFlags::kSame`, the fact that a shorthand is available doesn't mean that one is forced to use it.

Reply:
> `enum class` is not widely used in ROOT (even v7)

For ROOT7 we're trying to change that: unless good reasons, enums should be enum classes. And you have some good reasons here! But then we should have a common prefix for the constants, else `kSame` is too generic, see also your argument below ("What if we later decide to add a more general fuzzy comparison operator to RAxisBase" - which might have its own `kSame`...)

Reply:
I like this proposal! Switched to it in the latest commit.